### PR TITLE
UDF: query foreign data

### PR DESCRIPTION
**Pull request by CAJan93**
## Info

- **Description**

Add a description on how to query foreign data using UDFs

- **Notes**

Important for a customer

## For reviewers

- **Preview**

  [ Paste the preview link to the updated page(s) here. Edit this item after the preview site is ready. To find the updated pages, scroll down to locate and open the Amplify preview link and select the **upcoming** version of the documentation. ]

- **Key points**

  [ Parts that may need revision or extra consideration. ]

## Before merging

- [ ] I have checked the doc site preview, and the updated parts look good.

- [ ] I have acquired the approval from the owner (and optionally the reviewers) of the code PR and at least one tech writer (`CharlieSYH`, `emile-00`, & `hengm3467`).

### CODE REVIEW

**Comment by CAJan93** [comment]
**Open questions:**

- Can we create a generic UDF with which we can query any table in the PG server? 
- Can we improve error handling and tell the user in RisingWave that an error happened?

File: docs/sql/udf/udf-foreign-data.md [hunk 1]
```diff
@@ -0,0 +1,244 @@
+---
+id: udf-foreign-data
+slug: /udf-foreign-data
```
Comment (fuyufjh) [inline docs/sql/udf/udf-foreign-data.md:3]: It seems that this page is not properly linked from TOC
 https://pr-1337.d2fbku9n2b6wde.amplifyapp.com/docs/dev/udf-foreign-data/

File: docs/sql/udf/udf-foreign-data.md [hunk 2]
```diff
@@ -0,0 +1,244 @@
...
+  8 |  27 | Switch
+(7 rows)
+
+-- below query is equivalent to the above one
+-- but the WHERE clause is applied in RisingWave instead of filtering in postgres directly
+-- it therefore has worse performance than the one above
+select * from select_people_table_pg('SELECT * FROM people;'::VARCHAR) WHERE age > 25;
+```
+
+Please note that RisingWave will not show errors, but instead empty results, e.g.
```
Comment (fuyufjh) [inline docs/sql/udf/udf-foreign-data.md]:
```suggestion
Please note that if errors happend when evaluating the UDF, the results will be empty and error message could be found both in logs of compute nodes and the UDF server, e.g. 
```

**Review by fuyufjh — APPROVED** [review]
LGTM

**Pull request merged**